server/src/main/java/org/elasticsearch/action/get/TransportGetAction.java

```diff
@@ -125,13 +126,24 @@ protected void asyncShardOperation(GetRequest request, ShardId shardId, ActionLi
         IndexService indexService = indicesService.indexServiceSafe(shardId.getIndex());
         IndexShard indexShard = indexService.getShard(shardId.id());
         if (indexShard.routingEntry().isPromotableToPrimary() == false) {
+            // TODO: remove assertion (ES-9563)
             assert indexShard.indexSettings().isFastRefresh() == false
-                : "a search shard should not receive a TransportGetAction for an index with fast refresh";
+                || clusterService.state().getMinTransportVersion().onOrAfter(TransportVersions.FAST_REFRESH_RCO)
```

Comment:
I am not sure this assertion is safe. We risk another node seeing seeing that the min-transport-version advanced prior to this search node realizing it. And a GET request could thus arrive here ahead of the node knowing about the new  min-transport version.